prover/src/errors.rs

```diff
@@ -0,0 +1,34 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use std::{error, fmt};
+
+#[derive(Debug)]
+pub struct ProverError(String);
```

Comment:
I'm fine with making the error a string, but wouldn't it be easier and more explicit to make it its own enum, with multiple possible variants? Would certainly reduce the amount of `map_err` in the implementations.
We could do a `Custom/Other` error variant that is a string if we need that kind of flexibility.